checkpoint_ngfw/icon_checkpoint_ngfw/connection/connection.py

```diff
@@ -0,0 +1,152 @@
+import komand
+from .schema import ConnectionSchema, Input
+# Custom imports below
+from komand.exceptions import PluginException, ConnectionTestException
+import requests
+
+
+class Connection(komand.Connection):
+
+    def __init__(self):
+        super(self.__class__, self).__init__(input=ConnectionSchema())
+
+    def connect(self, params):
+        self.logger.info("Connect: Connecting...")
+        self.username = params.get(Input.USERNAME_PASSWORD, {}).get("username")
+        self.password = params.get(Input.USERNAME_PASSWORD, {}).get("password")
+
+        self.ssl_verify = params.get(Input.SSL_VERIFY)
+        self.server_ip = params.get(Input.SERVER)
+        self.server_port = params.get(Input.PORT)
+
+        self.server_and_port = f"https://{self.server_ip}:{self.server_port}"
+
+        self.get_sid()
+
+    def get_sid(self):
```

Comment:
I thought about this one for a while. I'm going to let it go. 

I could do this for all the actions that are post_and_publish, however, there's going to be information actions that just post. I hate writing a method to abstract post since it's pretty simple by its self.

Reply:
See show_access_rulebase...that's just a simple post.